components/auth/Join.tsx

```diff
@@ -95,18 +97,19 @@ const Join = () => {
           error={formik.touched.password ? formik.errors.password : undefined}
           onChange={formik.handleChange}
         />
-
-        <div className="form-control flex  flex-row items-center">
-          <div className="space-x-2">
-            <Checkbox type="checkbox" className='checkbox checkbox-primary checkbox-xs' onChange={(e) => {
-              formik.setFieldValue('agreeToTerms', e.target.checked);
-            }} />
-            <span className="checkbox-toggle"></span>
-          </div>
-          <label className="label">
-            <span className="label-text">Agree to <Link href='/terms-condition' className='text-primary' target="_blank">Terms and conditions</Link></span>
-          </label>
-        </div>
+        {process.env.NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL?.length !== 0 &&
```

Comment:
```suggestion
        {process.env.NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL &&
```
When NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL is not set (missing), the value is undefined.